main.go

```diff
@@ -205,13 +211,19 @@ func (s Step) createCodesignManager(inputs Inputs, xcodeMajorVersion int) (codes
 
 	var serviceConnection *devportalservice.AppleDeveloperConnection = nil
 	devPortalClientFactory := devportalclient.NewFactory(s.logger)
-	if authType == codesign.APIKeyAuth || authType == codesign.AppleIDAuth {
+	if inputs.BuildURL != "" && inputs.BuildAPIToken != "" {
```

Comment:
Could still skip creating connection if authType is `off`.

Reply:
The "off" option is handled at the beginning of `createCodesignmanager()`. In fact this whole function is not called if automatic code signing is off.